src/field_theory/normal.lean

```diff
@@ -124,42 +124,47 @@ end
 lemma alg_equiv.transfer_normal (f : E ≃ₐ[F] E') : normal F E ↔ normal F E' :=
 ⟨λ h, by exactI normal.of_alg_equiv f, λ h, by exactI normal.of_alg_equiv f.symm⟩
 
-lemma normal.of_is_splitting_field (p : F[X]) [hFEp : is_splitting_field F E p] :
-  normal F E :=
+-- seems to be causing a diamond in the below proof
+-- however, this may be a fluke and the proof below uses non-canonical `algebra` instances:
+-- when I replaced all the instances inside the proof with the "canonical" instances we have,
+-- I had the (unprovable) goal (of the form) `adjoin_root.mk f (C x) = adjoin_root.mk f X`
+-- for some `x, f`. So maybe this is indeed the correct approach and rewriting this proof is
+-- salient in the future, or at least taking a closer look at the algebra instances it uses.
+local attribute [-instance] adjoin_root.has_smul
```

Comment:
Just to check, this still applies after the PR split?

Reply:
Yes, it was the issue that I had to merge this file over

Reply:
In #18952 I've completely removed the `adjoin_root.has_smul` instance. It works in this proof, let's see what CI thinks.

Reply:
CI seems to be happy, so maybe there is no need to add the `has_smul` instance manually?

Reply:
I think it's needed for the `splitting_field` stuff; I can try remove it there and see?

Reply:
I don't think removing it is any better than locally de-instancing. The proof happens to work because the instance priorities find the `letI` before the distrb_smul.